File: lib/device.dart
class Device {
  String? ip;
  String? hostname;
  String? mac;
  String? vendor;
  double time = 0;

  Device(this.ip, this.hostname, this.mac, this.time, this.vendor);
  Device.fromJson(Map<String, dynamic> json)
      : ip = json["ip"],
        hostname = json["hostname"],
        mac = json["mac"],
        time = json["time"],
        vendor = json["vendor"];

  Map<String, dynamic> toJson() => {
        'hostname': hostname,
        'ip': ip,
        'time': time,
        'mac': mac,
        'vendor': vendor
      };
}

File: android/src/main/kotlin/com/myfreax/www/arp_scanner/subnet/Device.java
package com.myfreax.www.arp_scanner.subnet;

import java.net.InetAddress;

public class Device {
    public String ip;
    public String hostname;
    public String mac;
    public String vendor;
    public float time = 0;

    public Device(InetAddress ip) {
        this.ip = ip.getHostAddress();
        this.hostname = ip.getCanonicalHostName();
    }

    @Override
    public String toString() {
        return "Device{" +
                "ip='" + ip + '\'' +
                "vendor='" + vendor + '\'' +
                ", hostname='" + hostname + '\'' +
                ", mac='" + mac + '\'' +
                ", time=" + time +
                '}';
    }
}

